examples/convecting_plankton.jl

```diff
@@ -158,46 +217,75 @@ anim = @animate for (i, iteration) in enumerate(iterations)
-    p_lim = 1
+    w_levels = range(-w_lim, stop=w_lim, length=21)
+    w_lim < w_max && (w_levels = vcat([-w_max], w_levels, [w_max]))
 
-    w_levels = vcat([-w_max], range(-w_lim, stop=w_lim, length=21), [w_max])
-    p_levels = collect(range(p_min, stop=p_lim, length=20))
-    p_max > p_lim && push!(p_levels, p_max)
+    P_levels = collect(range(P_lims[1], stop=P_lims[2], length=20))
+    P_lims[1] > P_min && pushfirst!(P_levels, P_min)
+    P_lims[2] < P_max && push!(P_levels, P_max)
 
-    kwargs = (xlabel="x", ylabel="y", aspectratio=1, linewidth=0, colorbar=true,
+    kwargs = (xlabel="x (m)", ylabel="y (m)", aspectratio=1, linewidth=0, colorbar=true,
               xlims=(0, model.grid.Lx), ylims=(-model.grid.Lz, 0))
 
-    w_plot = contourf(xw, zw, w';
-                       color = :balance,
-                      levels = w_levels,
-                       clims = (-w_lim, w_lim),
-                      kwargs...)
-
-    p_plot = contourf(xp, zp, p';
-                       color = :matter,
-                      levels = p_levels,
-                       clims = (p_min, p_lim),
-                      kwargs...)
-
-    P_plot = plot(P, zp[:],
-                  linewidth = 2,
-                  label = nothing,
-                  xlims = (-0.2, 1),
-                  ylabel = "Averaged plankton",
-                  xlabel = "Plankton concentration")
-
-    plot(w_plot, p_plot, P_plot,
-         title=["Vertical velocity" "Plankton" "Averaged plankton"],
-         link = :y,
-         layout=(1, 3), size=(1700, 400))
+    w_contours = contourf(xw, zw, w';
+                          color = :balance,
+                          levels = w_levels,
+                          clims = (-w_lim, w_lim),
+                          kwargs...)
+
+    P_contours = contourf(xp, zp, P';
+                          color = :matter,
+                          levels = P_levels,
+                          clims = P_lims,
+                          kwargs...)
+
+    P_profile = plot(averaged_P, zp,
+                     linewidth = 2,
+                     label = nothing,
+                     xlims = (0.9, 1.3),
+                     ylabel = "z (m)",
+                     xlabel = "Plankton concentration (μM)")
+
+    flux_plot = plot(times ./ hour, buoyancy_flux_time_series,
+                     linewidth = 1,
+                     label = "Buoyancy flux time series",
+                     color = :black,
+                     alpha = 0.4,
+                     legend = :topright,
+                     xlabel = "Time (hours)",
+                     ylabel = "Buoyancy flux (m² s⁻³)",
+                     ylims = (0.0, 1.1 * buoyancy_flux_parameters.initial_buoyancy_flux))
+
+    plot!(flux_plot, times[1:i] ./ hour, buoyancy_flux_time_series[1:i],
+          color = :steelblue,
+          linewidth = 6,
+          label = nothing)
+
+    scatter!(flux_plot, times[i:i] / hour, buoyancy_flux_time_series[i:i],
+             markershape = :rtriangle,
```

Comment:
I suggest `markershape = :circle` because it feels a bit "weird" that the triangle doesn't align so that it's pointing along the tangent of the curve :)

Reply:
Heh true.